cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/BaseController.java

```diff
@@ -14,16 +20,62 @@
         subcommands = {
         CreateController.class,
         GetController.class})
-public class BaseController implements Runnable {
+public class BaseController implements Callable<Integer> {
 
     @Override
-    public void run() {
+    public Integer call() {
         // this code is executed if the user just runs the app
         CommandLine.usage(this, System.out);
+        return 0;
     }
 
     public static void main(String[] args) {
-        int exitCode = new CommandLine(new BaseController()).execute(args);
+        CommandLine cli = new CommandLine(new BaseController());
+
+        // picocli has a nice hidden feature: one can register a special exception handler and thus deal with
+        // exceptions occurring during the execution of a Callable, Runnable etc.
+        // when using Callables instead of Runnables, one can just forward _all_ exceptions to this exception handler
+        // this then allows for handling all exceptions centrally, which helps eliminate duplicate code and provide a
+        // consistent behavior, improving the overall UX on the commandline
+        // see https://picocli.info/apidocs/picocli/CommandLine.html#getExecutionExceptionHandler--
+        // and https://picocli.info/apidocs/picocli/CommandLine.html#execute-java.lang.String...-
+        cli.setExecutionExceptionHandler((ex, commandLine, parseResult) -> {
+            // TODO: different handling for different exceptions
+            if (ex instanceof IOException) {
+                Log.error("I/O error:", ex.getMessage());
+            } else if (ex instanceof CreationException) {
+                Log.error("Failed to create message:" + ex.getMessage());
+            } else if (ex instanceof UnsupportedOperationException) {
+                Log.error("Operation not supported/implemented:", ex.getMessage());
+            } else if (ex instanceof CredentialException) {
+                Log.error("Credentials error:", ex.getMessage());
+            } else if (ex instanceof IllegalStateException) {
+                // a little bit ugly, but it works
+                // the problem is in reactor.core.Exceptions
+                // it creates lambda class instances which are hard to test on...
+                if (ex.getMessage().toLowerCase().contains("retries exhausted")) {
+                    // now, if there is a cause and it's an InvalidTokenException, we can at least provide a hint
+                    // that it's likely a password issue
+                    if (ex.getCause() != null &&
```

Comment:
Seperation of concern, 
wenn man so eine tiefe Einrückung hat, ist es meist ein Indikator dafür, die Logik in eine Funktion auszulagern.